hypershift-operator/controllers/nodepool/conditions.go

```diff
@@ -852,3 +855,92 @@ func (r NodePoolReconciler) validPlatformConfigCondition(ctx context.Context, no
 	SetStatusCondition(&nodePool.Status.Conditions, *condition)
 	return nil, nil
 }
+
+func (r *NodePoolReconciler) supportedVersionSkewCondition(ctx context.Context, nodePool *hyperv1.NodePool, hcluster *hyperv1.HostedCluster) (*ctrl.Result, error) {
+	if hcluster.Status.Version == nil {
+		SetStatusCondition(&nodePool.Status.Conditions, hyperv1.NodePoolCondition{
+			Type:               hyperv1.NodePoolSupportedVersionSkewConditionType,
+			Status:             corev1.ConditionUnknown,
+			Reason:             hyperv1.NodePoolValidationFailedReason,
+			Message:            "HostedCluster version is not available yet, waiting for version information",
+			ObservedGeneration: nodePool.Generation,
+		})
+		return &ctrl.Result{}, fmt.Errorf("HostedCluster version is not available yet")
```

Comment:
An error returned from this function short circuits other condition checking. Do we always want to short circuit other conditions, or should the error be reported in the condition itself and continue to check the others?